### ISSUE TITLE
1.1.310 - Possible regression reportUndefinedVariable

### ISSUE THREAD

**Issue by sylvainmouquet** [microsoft/pyright#5169]
**Describe the bug**
```
commons/monitoring/datadog.py:69:12 - error: "tags" is not defined (reportUndefinedVariable)
commons/monitoring/datadog.py:80:17 - error: "_get_tags" is not defined (reportUndefinedVariable)
tests/test_datadog.py:5:40 - error: "_get_tags" is unknown import symbol (reportGeneralTypeIssues)

```
**To Reproduce**
```
def _get_tags(data: Dict[str, Any]):
    tags = {}
    for i, (k, v) in enumerate(data.items()):
        tags[f"payload.{k}"] = v
    return tags

def update_span(data: Dict[str, Any]):
  ...
            span.set_tags(
                _get_tags(data["payload"])
            )  # split all tags into payload.{key}
            ...

**Comment by erictraut** [comment]
Thanks for reporting the issue. I'm not able to repro the problem with the code you've supplied. It looks like you chopped out pieces of the code, so it's no longer valid. Could you provide a minimal, self-contained sample that exhibits the problem you're seeing?

Here's my attempt to modify your code so it's valid, but this doesn't exhibit any type checking errors:

```python
from typing import Any

def _get_tags(data: dict[str, Any]):
    tags = {}
    for i, (k, v) in enumerate(data.items()):
        tags[f"payload.{k}"] = v
    return tags

def update_span(data: dict[str, Any]):
    _get_tags(data["payload"])
```

**Comment by sylvainmouquet** [comment]
i close the issue, when i remove the code of the error https://github.com/microsoft/pyright/issues/5167 no warning are fired
